sources/packages/web/src/services/http/EducationProgramOfferingApi.ts

```diff
@@ -33,16 +38,42 @@ export class EducationProgramOfferingApi extends HttpBaseClient {
     }
   }
 
+  /**
+   * To get the offering summary
+   * @param locationId, location id
+   * @param programId, program id
+   * @param page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @param pageLimit, limit of the page if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @param searchName, user's name keyword to be searched
+   * @param sortField, field to be sorted
+   * @param sortOrder, order to be sorted
+   * @returns offering summary.
+   */
   public async getAllEducationProgramOffering(
     locationId: number,
     programId: number,
-  ): Promise<EducationProgramOfferingDto[]> {
+    page = DEFAULT_PAGE_NUMBER,
+    pageCount = DEFAULT_PAGE_LIMIT,
+    searchName?: string,
+    sortField?: OfferingSummaryFields,
+    sortOrder?: DataTableSortOrder,
+  ): Promise<PaginatedOffering> {
     try {
-      const response = await this.apiClient.get(
-        `institution/offering/location/${locationId}/education-program/${programId}`,
-        this.addAuthHeader(),
-      );
-      return response.data as EducationProgramOfferingDto[];
+      let URL = `institution/offering/location/${locationId}/education-program/${programId}?page=${page}&pageLimit=${pageCount}`;
+      if (searchName) {
+        URL = `${URL}&searchName=${searchName}`;
+      }
+      if (sortField && sortOrder) {
+        const sortDBOrder =
+          sortOrder === DataTableSortOrder.DESC
+            ? FieldSortOrder.DESC
+            : FieldSortOrder.ASC;
+        URL = `${URL}&sortField=${sortField}&sortOrder=${sortDBOrder}`;
```

Comment:
This logic is going to be repeated many times, can we have a small helper to build the URL with the sort parameters always in the same way?